flair/datasets/sequence_labeling.py

```diff
@@ -638,23 +674,31 @@ def _convert_lines_to_sentence(
         tokens: List[Token] = []
         filtered_lines = []
         comments = []
-        for line in lines:
-            # parse comments if possible
-            if self.comment_symbol is not None and line.startswith(self.comment_symbol):
-                comments.append(line)
-                continue
-
-            filtered_lines.append(line)
 
-            # otherwise, this line is a token. parse and add to sentence
-            token = self._parse_token(line, word_level_tag_columns, token)
-            tokens.append(token)
-        sentence: Sentence = Sentence(text=tokens)
+        sentence: Sentence
 
-        # check if this sentence is a document boundary
-        if sentence.to_original_text() == self.document_separator_token:
+        if len(lines) == 1 and lines[0].strip() == self.document_separator_line:
+            sentence = Sentence(text=[Token(lines[0])])
```

Comment:
`lines[0]` may not necessarily be the token text (some datasets have text in the second column). `self.text_column` stores in which column the text is.

Reply:
Plus, could you add a comment to explain this part of the refactor? Is this for speedup purposes?

Reply:
In CoNNL 2012, document boundaries are not marked by a specific token, but a line ("#end document" without any of the other columns -> treating this line as a token will result in a parsing error). I added this as a comment.

Regarding the index: lines[0] should be the only line in lines (with len 1), its not the first column in this line.